src/Module.ts

```diff
@@ -217,12 +196,12 @@ export default class Module {
 		if (ast) {
 			// prevent mutating the provided AST, as it may be reused on
 			// subsequent incremental rebuilds
-			this.ast = clone(ast);
+			this.ast = <Program>clone(ast);
 			this.astClone = ast;
 		} else {
 			// TODO what happens to comments if AST is provided?
-			this.ast = <any>tryParse(this, this.graph.acornParse, this.graph.acornOptions);
-			this.astClone = clone(this.ast);
+			this.ast = <Program>tryParse(this, this.graph.acornParse, this.graph.acornOptions);
```

Comment:
Continuing my previous question: Isn't `this.graph.acornParse` returning an `ESTree.Program`?

Reply:
Yes, but then we mutate this later through `enhance`, and TypeScript doesn't support mutating a type!